src/components/Pagination/Pagination.test.tsx

```diff
@@ -92,6 +92,28 @@ describe("Pagination", () => {
     expect(onChange).toHaveBeenCalledTimes(0);
   });
 
+  it("should not call onChange when input is greater than totalPages", () => {
+    const onChange = jest.fn();
+    const { getByDisplayValue } = renderPagination({
+      currentPage: 99,
+      totalPages: 100,
+      onChange,
+    });
+    const pageInput = getByDisplayValue("99");
+    fireEvent.input(pageInput, {
+      target: {
+        value: "100",
+      },
+    });
+    expect(onChange).toHaveBeenCalledTimes(1);
```

Comment:
is it me or is this calling to change to page 101  when max pages is 100?

Reply:
No it won't call it 
this should have been 2 if it is called again as it has been called for 100
Since the page number starts from 1 100 is also included